### PR TITLE
Allow reuse lookup to use current EOI partnership

### PR DESCRIPTION
**Pull request by ericaporter**
This PR is for [ticket](https://github.com/DFE-Digital/register-ects-project-board/issues/3060)
### Context

The original Reuse ticket is [here](https://github.com.mcas.ms/DFE-Digital/register-ects-project-board/issues/2205)

This PR clarifies and hardens the ECT “reuse previous partnership” flow.

Reuse selection now prefers current-year partnerships over historical ones. 

The wizard now correctly allows reuse when a school has a previous provider-led expression of interest (EOI) and the lead provider is available in the registration year, even if no partnership exists.

Specs have been expanded to cover partnership reuse, EOI reuse, and non-reusable edge cases.

### Guidance to review

First ever ECT at a school registered in 2025 contract period with LP1 & DP1.
-> Next ECTs should always show reuse choices screen (as there will always be a valid previous choice for 2025)

Last ECT in 2024 contract period registered with LP1 & DP1; and LP1& DP1 are still an available pairing in 2025.
-> Next ECTs registered in 2025 contract period should always show reuse choices screen (as there will always be a valid previous choice for 2025)

Last ECT in 2024 contract period registered with LP1 & DP1; and LP1& DP1 are not an available pairing in 2025.
-> Next ECT registered in 2025 contract period should not show reuse choices screen (as there is no valid previous choice for 2025).
-> But any next ECTs after that should always show reuse choices screen (as there will now be a valid previous choice for 2025).

### CODE REVIEW

File: app/wizards/schools/register_ect_wizard/start_date_step.rb [hunk 1]
```diff
@@ -12,7 +12,9 @@ def self.permitted_params
       end
 
       def next_step
-        return :working_pattern if past_start_date? || start_date_contract_period&.enabled?
+        return :cannot_register_ect_yet unless start_date_contract_period
```
[the remark this answers is not in the document]
Comment (ericaporter) [reply app/wizards/schools/register_ect_wizard/start_date_step.rb]: Now updated & added in a test. The intended behaviour isn't changing - but when I was debugging I noticed an edge case where if start_date_as_date was nil and past_start_date is true then it was still possible to progress past this step

[the remark this answers is not in the document]
Comment (thenapking) [reply app/wizards/schools/register_ect_wizard/start_date_step.rb]: Great, I'm also work on a change which has a very similar bit of logic. So I'll check for your edge case there too.
